lib/enum/nested.rb

```diff
@@ -18,21 +18,27 @@ def initialize(people_array)
   end
 
   def ages_sum
+    @people.map{|person| person.age}.reduce(:+)
   end
 
   def average_age
+    @people.map{|person| person.age}.reduce(:+) / people.length
   end
 
   def total_years_programming_experience_for_all_languages
+    @people.flat_map{|person| person.years_language_experience.values}.inject(:+)
   end
 
   def favorite_food_frequency
+    @people.flat_map {|person| person.favorite_foods}.each_with_object(Hash.new(0)) {|food, count| count[food] += 1}
```

Comment:
each_with_object takes a block with two parameters, but I wouldn't call them `|food, count|`. Instead they are `|food, new_hash|` where new_hash is the hash you created with Hash.new(0). It doesn't make sense to say `count[food] += 1` but it does make sense to create new keys in our hash with `new_hash[food] += 1`.